serialization/BSON.ksy

```diff
@@ -0,0 +1,165 @@
+        type: b15
+      - id: significand_hi
+        type: b49
+      - id: significand_lo
+        type: u8
+  object_id:
+    doc: "https://docs.mongodb.com/manual/reference/method/ObjectId/"
+    seq:
+      - id: epoch_time
+        type: u4
+        doc: "seconds since the Unix epoch"
+      - id: machine_id
+        type: u3
+      - id: process_id
+        type: u2
+      - id: counter
+        type: u3
+        doc: "counter, starting with a random value."
+  db_pointer:
+    seq:
+      - id: namespace
+        type: string
+      - id: id
+        type: object_id
+  binary:
+    doc: "The BSON \"binary\" or \"BinData\" datatype is used to represent arrays of bytes. It is somewhat analogous to the Java notion of a ByteArray. BSON binary values have a subtype. This is used to indicate what kind of data is in the byte array. Subtypes from zero to 127 are predefined or reserved. Subtypes from 128-255 are user-defined."
+    seq:
+      - id: len
+        type: s4
+      - id: subtype
+        type: u1
+        enum: subtype
+      - id: content
+        size: len
+    enums:
+      subtype:
+        0x00: generic #Generic binary subtype
+        0x01: function
+        0x02: binary_old #This used to be the default subtype, but was deprecated in favor of \x00. Drivers and tools should be sure to handle \x02 appropriately. The structure of the binary data (the byte* array in the binary non-terminal) must be an int32 followed by a (byte*). The int32 is the number of bytes in the repetition.
+        0x03: uuid_old #his used to be the UUID subtype, but was deprecated in favor of \x04. Drivers and tools for languages with a native UUID type should handle \x03 appropriately.
+        0x04: uuid
+        0x05: md5
+        0x80: user_defined #\x80-\xFF "User defined" subtypes. The binary data can be anything.
+  element:
+    seq:
+      - id: type
+        type: u1
+        enum: type
+      - id: name
+        type: cstring
+      - id: content
+        #if: "(type!=type::undefined && type!=type::null && type!=type::min_key && type!=type::max_key)"
+        type:
+          switch-on: type
+          cases:
+            'type::double': f8
+            'type::string': string
+            'type::document': document
+            'type::array': document
+            'type::binary': binary
+            'type::object_id': object_id
+            'type::boolean': u1
+            'type::utc_datetime': s8 #The int64 is UTC milliseconds since the Unix epoch.
+            'type::regex': regex
+            'type::db_pointer': db_pointer
+            'type::javascript': string
+            'type::symbol': string
+            'type::javascript_unscoped': code_with_scope
+            'type::int32': s4
+            'type::timestamp': u8 #Special internal type used by MongoDB replication and sharding. First 4 bytes are an increment, second 4 are a timestamp.
+            'type::int64': s8
+            'type::decimal128': f16
+    enums:
+      type:
```

Comment:
It turns out that most major problem is undeclared `u3` type. Is it supposed to "3-byte long unsigned integer?" Can we replace that with `type: b24` or it will lead to endianness problems?